File: 2277-count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.py
from collections import defaultdict


class Solution:
    def countPairs(self, nums: list[int], k: int) -> int:
        mp = defaultdict(lambda: defaultdict(int))
        ans = 0

        for i, num in enumerate(nums):
            mi = i % k
            if num in mp:
                for mj, count in mp[num].items():
                    if (mi * mj) % k == 0:
                        ans += count
            mp[num][mi] += 1

        return ans

File: 2277-count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countPairs = function (nums, k) {
    let count = 0;
    let n = nums.length;

    for (let i = 0; i < n - 1; i++) {
        for (let j = i + 1; j < n; j++) {
            if (nums[i] === nums[j] && ((i * j) % k === 0)) {
                count++;
            }
        }
    }

    return count;
};